liana-gui/src/app/state/transactions.rs

```diff
@@ -246,11 +259,27 @@ impl State for TransactionsPanel {
                     );
                 }
             }
-            _ => {
-                if let Some(modal) = &mut self.create_rbf_modal {
-                    return modal.update(daemon, _cache, message);
+            Message::View(view::Message::Export(ExportMessage::Open)) => {
+                if let TransactionsModal::None = &self.modal {
+                    self.modal = TransactionsModal::Export(ExportModal::new(daemon));
+                    if let TransactionsModal::Export(m) = &mut self.modal {
```

Comment:
nit: As this `if` condition should be true, perhaps you could store `ExportModal::new(daemon)` in a variable and then use it to call `launch()`.

Reply:
modal has to be stored in the state, but i can `.expect()` it if you prefer

Reply:
I had in mind something like:
```diff
diff --git a/liana-gui/src/app/state/transactions.rs b/liana-gui/src/app/state/transactions.rs
index ad90d5d3..a04b3829 100644
--- a/liana-gui/src/app/state/transactions.rs
+++ b/liana-gui/src/app/state/transactions.rs
@@ -261,10 +261,10 @@ impl State for TransactionsPanel {
             }
             Message::View(view::Message::Export(ExportMessage::Open)) => {
                 if let TransactionsModal::None = &self.modal {
-                    self.modal = TransactionsModal::Export(ExportModal::new(daemon));
-                    if let TransactionsModal::Export(m) = &mut self.modal {
-                        return m.launch();
-                    }
+                    let modal = ExportModal::new(daemon);
+                    let cmd = modal.launch();
+                    self.modal = TransactionsModal::Export(modal);
+                    return cmd;
                 }
             }
             Message::View(view::Message::Export(ExportMessage::Close)) => {
```
I think `.expect()` would also be fine or also to keep it as it is now.

You can remove the `mut` now following your change to the `launch()` method.

Reply:
i've dropped the `mut`